beater/mux.go

```diff
@@ -0,0 +1,220 @@
+package beater
+
+import (
+	"fmt"
+	"io"
+	"io/ioutil"
+	"strings"
+
+	"github.com/elastic/apm-server/processor"
+	"github.com/elastic/beats/libbeat/logp"
+
+	"compress/gzip"
+	"compress/zlib"
+	"encoding/json"
+	"errors"
+	"net/http"
+
+	"crypto/subtle"
+
+	"github.com/elastic/beats/libbeat/monitoring"
+)
+
+type chainable func(processor.Processor, Config, reporter) http.Handler
```

Comment:
let's call it something like `processorChain` / `processorHandler` / `processorHandlerChain` for more descriptiveness